services/device-registry-base/src/main/java/org/eclipse/hono/service/management/tenant/TrustedCertificateAuthority.java

```diff
@@ -374,4 +423,118 @@ public final TrustedCertificateAuthority setAuthIdTemplate(final String template
+     * Gets OCSP responder certificate which is used to verify OCSP response signature. If custom certificate is not
+     * set then issuing CA of client certificate is used instead.
+     *
+     * @return The DER encoded X.509 CA certificate used for OCSP signature verifications.
+     * @throws CertificateException if the certificate cannot be deserialized into a byte array.
+     */
+    public byte[] getOcspResponderCert() throws CertificateException {
+        if (ocspResponderCert != null) {
+            return ocspResponderCert;
+        } else if (ocspRevocationEnabled && cert != null) {
+            return cert.getEncoded();
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Sets OCSP responder certificate which is used to verify OCSP response signature. If custom certificate is not
+     * set then issuing CA of client certificate is used instead.
+     *
+     * @param ocspResponderCert The Custom DER encoded X.509 CA certificate used for verification of OCSP response
+     *                          signature or null to use client certificate issuing CA instead.
+     * @return A reference to this for fluent use.
+     */
+    public TrustedCertificateAuthority setOcspResponderCert(final byte[] ocspResponderCert) {
+        this.ocspResponderCert = ocspResponderCert;
+        return this;
+    }
+
+    /**
+     * Gets whether nonce extension should be sent in OCSP request. Nonce is important to avoid replay attacks but can
+     * increase resource usage. It is disabled by default.
+     *
+     * @return True if nonce extension is enabled.
+     */
+    public boolean isOcspNonceEnabled() {
+        return ocspNonceEnabled;
+    }
+
+    /**
+     * Sets whether nonce extension should be sent in OCSP request. Nonce is important to avoid replay attacks but can
+     * increase resource usage. It is disabled by default.
+     *
+     * @param ocspNonceEnabled True to enable nonce extension in OCSP requests.
+     * @return A reference to this for fluent use.
+     */
+    public TrustedCertificateAuthority setOcspNonceEnabled(final boolean ocspNonceEnabled) {
+        this.ocspNonceEnabled = ocspNonceEnabled;
+        return this;
+    }
+
+    /**
+     * Gets whether only end (leaf) certificate revocation should be checked. It is set to false by default.
+     *
+     * @return True if only end (leaf) certificate revocation should be checked.
+     */
+    public boolean isCheckEndEntityOnly() {
+        return checkEndEntityOnly;
+    }
+
+    /**
+     * Sets whether only end (leaf) certificate revocation should be checked. It is set to false by default.
+     *
+     * @param checkEndEntityOnly True if only end (leaf) certificate revocation should be checked.
+     * @return A reference to this for fluent use.
+     */
+    public TrustedCertificateAuthority setCheckEndEntityOnly(final boolean checkEndEntityOnly) {
+        this.checkEndEntityOnly = checkEndEntityOnly;
+        return this;
+    }
```

Comment:
I'd rather not make this configurable for the time being and always only check end certs.

Reply:
Removed option